### PR TITLE
Update contribution guidelines

### PR DESCRIPTION
**Pull request by windsource**
Issues: #36

# Definition of Done

The PR shall be merged **only if all of following items were checked.** In case an item is not applicable as described, please provide a short explanation right after the item in question. Separate the explanation by a semicolon and write it as bold text like **; requirements are handled in issue #xyz**:

- [x] documentation of all modules `/*/doc/swdesign` is up-to-date
- [x] requirements are up-to-date and requirements for new features have been added and mapped to source and test
- [x] conform to [unit verification strategy](https://eclipse-ankaios.github.io/ankaios/main/development/unit-verification/)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Contributing
 
 Welcome to the Ankaios community. Start here for info on how to contribute and help improve our project.
```
Comment (lingnoi) [inline CONTRIBUTING.md:3]:
```suggestion
Welcome to the Ankaios community. Start here for info on how to contribute and help to improve our project.
```

File: doc/docs/development/requirement-tracing.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
-# Requirements tracing
+# Requirement tracing
 
-The Eclipse Ankaios project provides requirements tracing using the [Open Fast Trace](https://github.com/itsallcode/openfasttrace) requirement tracing suite. The dev container already includes an `oft` executable and supports the `oft` command. For convenience, you can run the `tools/generate_oft_html_report.sh` script to automatically generate an HTML report showing the current coverage state. The script automatically includes all src and doc folders in the root folder. As long as you name your folders accordingly they will be processed by `oft`. 
+## Introduction
 
-For details on the `oft` tool, please consult the [user documentation](https://github.com/itsallcode/openfasttrace/blob/main/doc/user_guide.md) or execute `oft help`.
+The Eclipse Ankaios project provides requirements tracing using the [OpenFastTrace](https://github.com/itsallcode/openfasttrace) requirement tracing suite. The dev container already includes the required tooling. To generate a requirement tracing report call:
```
Comment (lingnoi) [inline doc/docs/development/requirement-tracing.md:5]:
```suggestion
The Eclipse Ankaios project provides requirement tracing using the [OpenFastTrace](https://github.com/itsallcode/openfasttrace) requirement tracing suite. The dev container already includes the required tooling. To generate a requirement tracing report call:
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (krucod3) [reply CONTRIBUTING.md:3]: I would leave it as it is

File: doc/docs/development/requirement-tracing.md [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+
+```shell
+tools/generate_oft_html_report.sh
+```
+
+Afterwards the HTML report is available in `build/req/req_tracing_report.html` showing the current coverage state..
```
Comment (krucod3) [inline doc/docs/development/requirement-tracing.md:11]:
```suggestion
Afterwards the HTML report is available under `build/req/req_tracing_report.html` and shows the current coverage state.
```

File: doc/docs/development/requirement-tracing.md [hunk 4]
```diff
@@ -1,5 +1,27 @@
...
+
+For details on the OpenFastTrace tool, please consult the [user documentation](https://github.com/itsallcode/openfasttrace/blob/main/doc/user_guide.md) or execute `oft help`.
```
Comment (krucod3) [inline doc/docs/development/requirement-tracing.md:13]:
```suggestion
For details on the OpenFastTrace tool, please consult [OFT's user documentation](https://github.com/itsallcode/openfasttrace/blob/main/doc/user_guide.md) or execute `oft help`.
```

File: doc/docs/development/requirement-tracing.md [hunk 5]
```diff
@@ -1,5 +1,27 @@
...
+
+* Design (`**/doc/README.md`)
+* Implementations (`**/src/**`)
+* Tests (`**/src/**`, `tests/**`)
+
+So for new features 
+
+* New requirements need to be added in the design or existing requirements need to be modified (type `swdd`)
+* Mark the parts in the source code that actually implement the design requirement (type `impl`)
+* Mark the tests that test the design and implementation (type `utest`, `itest` or `stest`)
```
Comment (krucod3) [inline doc/docs/development/requirement-tracing.md:27]:
```suggestion
* Mark the parts in the source code that actually implement the design requirement using the type `impl`, e.g., `// [impl->swdd~this-is-a-requirement~1]`
* Mark the tests that check the implementation of the design with one of the types `utest`, `itest` or `stest` depending on the type of the test, e.g., `// [utest->swdd~this-is-a-requirement~1]` for a unit test
```

**Review by krucod3 — CHANGES_REQUESTED** [review]
Look good, just documented some small findings.

File: doc/docs/development/requirement-tracing.md (hunk 3 above)
Comment (windsource) [reply doc/docs/development/requirement-tracing.md:11]: OK

File: doc/docs/development/requirement-tracing.md (hunk 4 above)
Comment (windsource) [reply doc/docs/development/requirement-tracing.md:13]: OK

File: CONTRIBUTING.md (hunk 1 above)
Comment (windsource) [reply CONTRIBUTING.md:3]: I have asked the technical writer of Eclipse SDV for a recommendation.

Comment (craigmateo) [reply CONTRIBUTING.md:3]: "help improve", but either is fine.

Comment (windsource) [reply CONTRIBUTING.md:3]: So let's keep it as it is.

**Review by krucod3 — APPROVED** [review]
LGTM 👍

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -15,13 +16,15 @@ pull request to integrate them.
...
+- **Requirements**: Make sure that requirements are created for new features and those are [traced in the code and tests](https://eclipse-ankaios.github.io/ankaios/main/development/requirement-tracing/).
+- **Rust coding guidelines**: Make sure to follow the [Rust coding guidelines for this project](https://eclipse-ankaios.github.io/ankaios/main/development/rust-coding-guidelines/).
+- **Unit verification strategy**: Unit tests have been created according to the [unit verification strategy](https://eclipse-ankaios.github.io/ankaios/main/development/unit-verification/).
 
 ## Communication
 
-Please join our [developer mailing list](https://accounts.eclipse.org/mailing-list/ankaios-dev) for up to date information.
+Please join our [developer mailing list](https://accounts.eclipse.org/mailing-list/ankaios-dev)
+for up to date information or use the Ankaios
+[discussion forum](https://github.com/eclipse-ankaios/ankaios/discussions).
```
Comment (krucod3) [inline CONTRIBUTING.md:29]:
```suggestion
Please join our [developer mailing list](https://accounts.eclipse.org/mailing-list/ankaios-dev) for up to date information or use the Ankaios [discussion forum](https://github.com/eclipse-ankaios/ankaios/discussions).
```

**Pull request merged**